lib/gitsh/input_strategies/interactive.rb

```diff
@@ -48,6 +49,19 @@ def read_command
         retry
       end
 
+      def read_continuation
+        input = begin
+          line_editor.readline(CONTINUATION_PROMPT, true)
```

Comment:
Passing `true` here results in each line of the command having it's own entry in the history. GNU Bash combines multi-line commands into a single history entry. That's probably a worthwhile enhancement, but maybe it should be a separate PR.